src/styles/main.scss

```diff
@@ -1,3 +1,59 @@
 body {
   margin: 0;
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 100px;
+  color: white;
+}
+
+.blocks {
+  display: grid;
+
+  @media (max-width: 599px) {
+    & {
+      grid-template-columns: repeat(auto, 300px);
+    }
+  }
+
+  @media (min-width: 600px) and (max-width: 899px) {
+    & {
+      grid-template-columns: repeat(2, minmax(300px, 1fr));
+    }
+  }
+
+  @media (min-width: 899px) {
+    & {
+      grid-template-columns: repeat(3, minmax(300px, 1fr));
+    }
+  }
+
+  &__block {
+    display: flex;
+    height: 300px;
+    align-items: center;
+    justify-content: center;
+  }
+
+  &__block--1 {
+    background-color: rgb(255, 0, 0);
+  }
+
+  &__block--2 {
+    background-color: rgb(204, 0, 0);
+  }
+
+  &__block--3 {
+    background-color: rgb(153, 0, 0);
+  }
+
+  &__block--4 {
+    background-color: rgb(102, 0, 0);
+  }
+
+  &__block--5 {
+    background-color: rgb(51, 0, 0);
+  }
+
+  &__block--6 {
+    background-color: rgb(0, 0, 0);
+  }
```

Comment:
You can do this with less code. Create variable and put your colors like this - &variable {
  rgb(255, 0, 0) 1,
  rgb(204, 0, 0) 2,
  rgb(153, 0, 0) 3,
  rgb(100, 0, 0) 4,
  rgb(51, 0, 0) 5,
  rgb(0, 0, 0) 6; 
  }

Then write it like this in your .block class
  @each $color, $order in $variable{
      &:nth-child(#{$order}) {
        background-color: $color;
      }

Reply:
fixed)